### PR TITLE
controller: Update routes for config APIs (#379)

### PR DESCRIPTION
**Pull request by nickbp**
- Updates application ingress to route `<cluster>/api/*` to the config-api pod (was: `config.<cluster>/api/*`)
- Moves cortex alertmanager and ruler APIs into the config-api pod
- Updates the config-api pod with alertmanager and ruler endpoints to be proxied
- Refactors auth and proxy handling a bit for reuse across services, while adding custom path override support when routing requests

### CODE REVIEW

**Comment by nickbp** [comment]
Still needs testing, don't merge yet

**Comment by MatApple** [comment]
> Still needs testing, don't merge yet

Feel free to add [wip] in the title until you're ready for it to be merged. This will prevent anyone from merging even if CI is green.

**Comment by nickbp** [comment]
Oh right good point

File: packages/controller/src/resources/app/api.ts
Comment (MatApple) [inline packages/controller/src/resources/app/api.ts:105]: I like this much better. We should lean into this convention and use it for all our deployments

**Review by MatApple — APPROVED** [review]
LGTM 👍

**Comment by nickbp** [comment]
Looks like things are routing correctly, checked the redirected endpoints as follows and the routing/responses looked reasonable:
```
CLUSTER=<cluster-name-here>
AUTH_HEADER="Authorization: Bearer $(cat tenant-api-token-system)"

curl -v -k -H $AUTH_HEADER https://$CLUSTER.opstrace.io/api/v1/ruler/ring
# level=debug msg="Redirecting req=/api/v1/ruler/ring to dest=ruler.cortex.svc.cluster.local/ruler/ring"
# response: HTML table of ruler instances

curl -v -k -H $AUTH_HEADER https://$CLUSTER.opstrace.io/api/v1/rules
# level=debug msg="Redirecting req=/api/v1/rules to dest=ruler.cortex.svc.cluster.local/api/v1/rules"
# response: "no rule groups found"

curl -v -k -H $AUTH_HEADER https://$CLUSTER.opstrace.io/api/v1/alerts
# level=debug msg="Redirecting req=/api/v1/alerts to dest=alertmanager.cortex.svc.cluster.local/api/v1/alerts"
# response: "the Alertmanager is not configured"

curl -v -k -H $AUTH_HEADER https://$CLUSTER.opstrace.io/api/v1/alertmanager
# level=debug msg="Redirecting req=/api/v1/alertmanager to dest=alertmanager.cortex.svc.cluster.local/api/v1/alertmanager"
# response: "the Alertmanager is not configured"

curl -v -k -H $AUTH_HEADER https://$CLUSTER.opstrace.io/api/v1/multitenant_alertmanager/status
# level=debug msg="Redirecting req=/api/v1/multitenant_alertmanager/status to dest=alertmanager.cortex.svc.cluster.local/multitenant_alertmanager/status"
# response: "Cortex Alertmanager Status" HTML page
```

**Comment by spahl** [comment]
This looks correct, thank you.

**Pull request merged**
